addons/website/static/src/snippets/s_social_media/options.js

```diff
@@ -3,34 +3,73 @@
+        this.links = [...this.$target[0].querySelectorAll(':scope > a')]
+            .map(el => ({id: generateHTMLId(), el: el}));
         await this._super(...arguments);
-        let websiteId;
-        this.trigger_up('context_get', {
-            callback: function (ctx) {
-                websiteId = ctx['website_id'];
-            },
-        });
-        // Fetch URLs for db links.
-        [this.dbSocialValues] = await this._rpc({
-            model: 'website',
-            method: 'read',
-            args: [websiteId, ['social_facebook', 'social_twitter', 'social_youtube',
-                'social_instagram', 'social_linkedin', 'social_github']],
-        });
-        delete this.dbSocialValues.id;
+        if (!dbSocialValues) {
+            let websiteId;
+            this.trigger_up('context_get', {
+                callback: function (ctx) {
+                    websiteId = ctx['website_id'];
+                },
+            });
+            // Fetch URLs for db links.
+            [dbSocialValues] = await this._rpc({
+                model: 'website',
+                method: 'read',
+                args: [websiteId, ['social_facebook', 'social_twitter', 'social_youtube',
+                    'social_instagram', 'social_linkedin', 'social_github']],
+            });
+            delete dbSocialValues.id;
+        }
+        for (const socialMedia of Object.keys(dbSocialValues)) {
+            const mediaName = socialMedia.split('social_').pop();
+            if (!this.$target[0].querySelector(`:scope > a[href="/website/social/${mediaName}"]`)) {
+                const anchorEl = this.$target[0].querySelector(':scope > a').cloneNode(true);
+                this._removeSocialMediaClasses(anchorEl);
+                const iEl = anchorEl.querySelector('i');
+                anchorEl.classList.add(`s_social_media_${mediaName}`);
+                iEl.classList.add(`fa-${mediaName}`);
+                anchorEl.href = `/website/social/${mediaName}`;
+                this.links.push({
+                    id: generateHTMLId(),
+                    el: anchorEl,
+                });
+            }
+        }
+        this.justStarted = true;
+    },
+    /**
+     * @override
+     */
+    onBuilt() {
+        for (const anchorEl of this.$target[0].querySelectorAll(':scope > a')) {
+            const mediaName = anchorEl.href.split('/website/social/').pop();
+            if (mediaName && !dbSocialValues[`social_${mediaName}`]) {
+                // Hide social media without value in DB.
+                anchorEl.classList.add('d-none');
+            }
+        }
+        if (this.$target[0].querySelectorAll(':scope > a:not(.d-none)').length < 1) {
+            // Ensure we do not drop a blank block.
+            const titleEl = this.$target[0].querySelector('.s_social_media_title');
+            titleEl.textContent = _t('Click here to add social media');
+            titleEl.classList.remove('d-none');
+        }
```

Comment:
Cool to have think about that :+1: 
But don't you think it would be better to add something that is only there during the edition?
Something like what we have where there is no image in the carousel
![image](https://user-images.githubusercontent.com/30048408/158796356-cfd61591-c8c4-48f5-baa0-9d282e5a08e4.png)

Right now the solution you suggest seems to lead to a poor UX:
1. The text is silently edited from `Social Media` to `Click here to add social media`, so after adding a social media, you have to change that text
2. It is probably not at the right place: Drag & drop the snippet, remove all links, the message do not show while it should, you can never click back on the social media
3. It is barely noticable, it looks like regular text.. Lost in the middle of a big pages, while a button like the carousel would make the user look at it
![image](https://user-images.githubusercontent.com/30048408/158797230-5db8eb65-ce0c-4f7c-a3fa-d30476958fff.png)

Anyway, just make it work the same way as we do elsewhere:
- Temporary DOM element during the edition, **it should not be saved**
- Make it visually different from regular content, it should feel like a configuration stuff
- It should appear when entering edit mode if the element is "empty" **and** if you empty the element during the edition